CMakeLists.txt

```diff
@@ -8,19 +8,60 @@ project(bergamot_translator CXX C)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(BUILD_ARCH native CACHE STRING "Compile for this CPU architecture.")
 
-# Custom CMake options to compile marian (a 3rd party submodule) for this project
-option(COMPILE_CUDA "Compile GPU version" OFF)
-option(USE_SENTENCEPIECE "Download and compile SentencePiece" ON)
-option(USE_STATIC_LIBS "Link statically against non-system libs" ON)
-option(USE_MKL "Compile with MKL support" ON)
+# Project specific cmake options
+option(COMPILE_WASM "Compile for WASM" OFF)
+option(COMPILE_THREAD_VARIANT "Compile with thread support" OFF)
+SET(PACKAGE_DIR "" CACHE STRING "Directory including all the files to be packaged (pre-loaded) in wasm builds")
 
-execute_process(COMMAND git submodule update --init --recursive --no-fetch
-                WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+# Set marian (3rd party submodule) cmake options to compile for this project
+SET(COMPILE_CUDA OFF CACHE BOOL "Compile GPU version")
+SET(USE_SENTENCEPIECE ON CACHE BOOL "Download and compile SentencePiece")
+SET(USE_STATIC_LIBS ON CACHE BOOL "Link statically against non-system libs")
+SET(USE_MKL OFF CACHE BOOL "Compile with MKL support")
+SET(COMPILE_DECODER_ONLY ON CACHE BOOL "Compile marian-decoder only")
+SET(COMPILE_WITH_PTHREADS OFF CACHE BOOL "Compile with pthreads support")
+SET(USE_WASM_COMPATIBLE_BLAS ON CACHE BOOL "Compile with a WASM compatible blas for decoder only builds")
+SET(COMPILE_LIBRARY_ONLY ON CACHE BOOL "Build only the Marian library and exclude all executables.")
+SET(COMPILE_WITHOUT_EXCEPTIONS ON CACHE BOOL "Compile without exceptions")
+if(COMPILE_WASM)
+  # Set WORMHOLE to ON for marian whenever compiling for wasm platform
+  SET(WORMHOLE ON CACHE BOOL "Use WASM wormhole in intgemm https://bugzilla.mozilla.org/show_bug.cgi?id=1672160")
+endif()
```

Comment:
What is `COMPILE_WITH_PTHREADS`, `COMPILE_THREAD_VARIANT` seems to be what's activating the no threads source through `ifdef`s. This is a bit confusing with `WITH_PTHREADS` being the `ifdef` check.

Reply:
COMPILE_WITH_PTHREADS is a marian specific cmake option. COMPILE_THREAD_VARIANT is specific to bergamot-translator project. We want to keep separate cmake options for both of them to control them individually.

Reply:
Given that you said threads are on the roadmap, why are we spending so much time breaking threads?